### ISSUE TITLE
Can't create more than one credit memo if order's grand total is 0

### ISSUE THREAD

**Issue by mushketer888** [magento/magento2#33237]
### Preconditions (*)
Magento 2.3.3

### Steps to reproduce (*)
1. Crete order with 100% discount on all items and shipping. (Cart rule promo code can be used for that) Grand total (including shipping) must be 0. Order must contain product's quantity atleast as 2
2. Create credit memo and refund order with "Qty to Refund" = 1. ("Return to stock" doesn't affect on this issue)
![2021-06-1413:35:50](https://user-images.githubusercontent.com/2189038/121879583-b3d4b100-cd15-11eb-94dc-71bda1d13e59.png)

### Expected result (*)
1. Be able to refund left ordered/invoiced items (to fully refund order/items)

### Actual result (*)
1. Magento 2 doesn't give ability to create additional credit memo

In my opinion the problem lies in Magento\Sales\Model\Order.php canCreditmemo()
and in canCreditmemoForZeroTotal() see [HERE](https://github.com/magento/magento2/blob/1990aec2ba665eaecaf1af99c132c91eec606be4/app/code/Magento/Sales/Model/Order.php#L742)
```
//check if total paid is less than grandtotal
$checkAmtTotalPaid = $totalPaid <= $this->getGrandTotal();
```

So if totalPaid ==0 and grandTotal ==0 then canCreditmemoForZeroTotal==false

But more precise maybe this code:
`$paidAmtIsRefunded = $this->getTotalRefunded() == $totalPaid && $creditmemos;`
totalRefunded==0, totalPaid==0 and has credit memos

Is there some patch or workaround for that?

- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._

**Comment by engcom-Echo** [comment]
Hi @mushketer888 

We are not able to re-produce the issue based on steps provided. Can you please verify the same on latest magento instance that is 2.4-develop and let us know if you get this issue there.

Thanks

**Comment by mushketer888** [comment]
@magento give me 2.4-develop instance

**Comment by mushketer888** [comment]
I was able to replicate this issue:
![2021-06-1515:34:37](https://user-images.githubusercontent.com/2189038/122053223-39785f80-cdef-11eb-8f51-d64fa316d82a.png)
Order #000000004
Only one credit memo can be created!

**Comment by engcom-Delta** [comment]
Hi @mushketer888,

Tried to reproduce the issue but step 2 is not clear , Hence Kindly provide steps for Creating credit memo.

Therefore , Added the label 'Needs update'.

Thanks

**Comment by mushketer888** [comment]
Hi @engcom-Delta 
The step 2 is just to create credit memo for order which has discount on all items and shipping with amount of 100% but with less product's quantity that order/invoice has

For example:
order 2 same products (qty=2) with discount of 100% (with shipping discount of 100% too)
create creditmemo but refund only 1 product
then try to create one more (second) creditmemo to refund lasting product item - you won't be able to do it
so basically if order's grand total = 0 then you can create only one credit memo

**Comment by engcom-Delta** [comment]
Hi @mushketer888,

I tried to reproduce the issue on Magento 2.4.2 as per provided steps & comments.
in fact, i have not able to create credit memo for the order which has 100% discount( cart items + shipping).
Kindly find the attached screenshots for reference. 
![qnty_2_discount100percent](https://user-images.githubusercontent.com/51681379/125306761-189f1d80-e34d-11eb-99c0-b81133f6a7af.jpg)
![quantity_2_100percentdisc](https://user-images.githubusercontent.com/51681379/125306794-22288580-e34d-11eb-97da-6e73ff09ca54.jpg)
Please let me know how to proceed further on this issue.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Delta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-42822` were created

**Issue Available**: @engcom-Delta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by simonmaass** [comment]
can confirm this for 2.4.4

**Comment by simonmaass** [comment]
any chance this will get resolved?

**Comment by nishanthan1989** [comment]
any update on this ?

**Comment by engcom-Bravo** [comment]
Hi @mushketer888,

Thanks for your reporting and collaboration.

We have verified the issue in Latest 2.4-develop instance and the issue is no more reproducible.Kindly refer the screenshots.

<img width="1156" alt="Screenshot 2024-06-13 at 09 56 06" src="https://github.com/magento/magento2/assets/51680745/9e84fdde-2188-4b12-a683-f91e5550d109">


<img width="1156" alt="Screenshot 2024-06-13 at 09 56 45" src="https://github.com/magento/magento2/assets/51680745/f46d9530-ceee-447c-8f49-cbda90331384">


We are able to refund left ordered/invoiced items (to fully refund order/items)

Hence We are closing this issue.

Thanks.
